packages/app/src/server/routes/api/cortex/index.ts

```diff
@@ -62,16 +62,30 @@ cortexProxy.on("proxyRes", (proxyRes, req, res) => {
        * - https://github.com/cortexproject/cortex/blob/82b32ec65ed16920e6053ac7fb748c42e3cae452/pkg/compactor/compactor_http.go#L16-L25
        */
       const body = chunks.join();
-      const parsedHTML = parse(body);
-      const errorMessage = parsedHTML.querySelector("html body p")?.text;
-
-      /* setting the status code as 412, as usually HTML error responses
-       * of cortex are due to misconfigurations
-       */
-      res.writeHead(errorMessage ? 412 : 500, {
-        "content-type": "text/plain"
-      });
-      res.end(errorMessage ?? "An unknown error occured.");
+      try {
+        const parsedHTML = parse(body);
+        const errorMessage = parsedHTML.querySelector("html body p")?.text;
+
+        if (!errorMessage) {
+          throw new Error("Could not extract error message");
+        }
+
+        /* setting the status code as 412, as usually HTML error responses
+         * of cortex are due to misconfigurations
+         */
+        res.writeHead(412, {
```

Comment:
btw, I know this was already in your previous patch but I didn't really notice and would like to comment on that now: all 4xx HTTP responses are supposed to mean that something is wrong with the _request_. The 412 response has a very specific meaning with respect to the request headers. So, semantically, it's a bit of an odd choice to communicate this problem in this kind of response.

Things like that are fine to be wrapped in a 500 response. Then of course one should find a way to allow the client to distinguish a generic "unexpected error" in the backend, from a specific (and potentially expected) error in the backend. In APIs this is often done by putting error detail into a JSON document into the response body. One can then do the following on the client side:
- get a 500 response. inspect the body. is this JSON?
    - yes: 
        - is this JSON of expected format containing err information?
           - yes: the backend communicated an expected, specific error to us -- we know exactly what that is and can handle it in a specific way

If any of the answers is "no": something unexpected happened in the backend and we handle that differently

Reply:
(just food for thought, this is your playground here :))

Reply:
here is an example in our own code base where we use this kind of appraoch to distinguish 404 responses from 404 responses via said method:
https://github.com/opstrace/opstrace/blob/9d683c95f934788cd38860d4df148310f0489cbf/lib/dns/src/opstrace.ts#L345

(in this case not via a specific JSON structure, but via presence of a specific error code which serves as a contract/convention)

Reply:
> In APIs this is often done by putting error detail into a JSON document into the response body.

Do we already have a standardised JSON format for our "internal" error transportation via HTTP? If we go this route, it might make sense to keep them consistent across the app.

Reply:
I just added one for now - if that works great, if not, I am happy to change it.